File: backend/mr-db-sql/src/main/java/io/featurehub/db/utils/UUIDIdGenerator.java
package io.featurehub.db.utils;

import io.ebean.config.IdGenerator;

import java.util.UUID;

public class UUIDIdGenerator implements IdGenerator  {
  @Override
  public Object nextValue() {
    return UUID.randomUUID().toString();
  }

  @Override
  public String getName() {
    return "uuidStr";
  }
}

File: backend/common-db/src/main/kotlin/io/featurehub/app/db/utils/UUIDIdGenerator.kt
package io.featurehub.app.db.utils

import io.ebean.config.IdGenerator
import java.util.*

class UUIDIdGenerator : IdGenerator {
  override fun nextValue(): Any {
    return UUID.randomUUID().toString()
  }

  override fun getName(): String {
    return "uuidStr"
  }
}
